zebedee/client.go

```diff
@@ -0,0 +1,302 @@
+package zebedee
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ONSdigital/dp-frontend-models/model"
+	"github.com/ONSdigital/dp-frontend-models/model/datasetLandingPageStatic"
+	"github.com/ONSdigital/go-ns/log"
+	"github.com/ONSdigital/go-ns/zebedee/models"
+	"github.com/c2h5oh/datasize"
+)
+
+// Client is an interface for communication to a zebedee server:
+// https://github.com/ONSdigital/zebedee
+//
+// Get will return a response from the server as a byte array, usually in a json
+// format.
+//
+// GetLanding will return a static dataset landing page struct populated with the
+// data from a given path in zebedee.
+type Client interface {
```

Comment:
also based on `interfaces in, structs out` do we even need to define the Client interface here?